### DISCUSSION TITLE
based freenet CHK and SSK decentralized IP address, Abolish ipv4 and ipv6

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [freenet/freenet-core#906] [General]
THE IDEAL PATH
What if we replace the controlled and surveilled IP addresses with new internet addresses that are based on CHK and SSK? These addresses will inherit all CHK and SSK features, i.e., they will be purely decentralized, secure, future-proof, robust, anonymous, unhackable, controlled by no single authority and many more.

Is it just a dream? For now. If this could be true we would be changing the internet as we know it.

* CHK and SSK address + CHK and SSK DNS, access to freenet domain

win10 network adapters : Enter the CHK and SSK address
Wi-Fi  DHCP Static : Enter the CHK and SSK address

---
IPv4 and IPv6 are centralized. They are controlled by national companies, etc. 
![](https://www.apnic.net/wp-content/uploads/policy-environment_obsolete/images/ipv4_hierarchy.png)

Try to Google “Who controls IP addresses?” You’ll promptly get “IANA: the Internet Assigned Numbers Authority.” IANA is the top authority behind IP address allocation and assignment. There are five different regional internet registries (RIR) with jurisdiction under the IANA.

As a matter of fact, as an individual or a normal internet user you cannot request IP addresses directly from IANA or one of the five RIRs, but only from internet service providers, such as the services offered by mobile or telecom operators.

* True decentralization, no official singular Foundation, Committee, Corporation, or entities in permanent unitary control of the protocol.

**Comment by Julian-Dumitrascu** [comment]
I like your message and your profile. When you (or anybody else) want to talk about this topic, I'd gladly find time for this.
